weaviate/connection/rest.go

```diff
@@ -53,6 +54,39 @@ func NewConnection(scheme string, host string, httpClient *http.Client, headers
 	return connection
 }
 
+// WaitForWeaviate waits until weaviate is started up and ready
+func (con *Connection) WaitForWeaviate(startupTimeout time.Duration) error {
+	if startupTimeout < 0 {
+		return errors.New("'startupTimeout' needs to be a time.Duration larger than zero")
+	}
+	ticker := time.NewTicker(time.Second)
+	startTime := time.Now()
+	for {
+		t := <-ticker.C
+		ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second)
+		response, err := con.RunREST(ctx, "/.well-known/ready", http.MethodGet, nil)
+		var isReady bool
+		switch {
+		case err != nil:
+			isReady = false
+		case response.StatusCode == 200:
+			isReady = true
+		default:
+			isReady = false
+
+		}
+
+		cancelFunc()
+		if isReady {
+			return nil
+		}
+		if t.After(startTime.Add(startupTimeout)) {
+			return fmt.Errorf("weaviate did not start up in %d seconds. Either the Weaviate URL %q is wrong or Weaviate did not start up in the interval given in 'startupTimeout'", startupTimeout, con.basePath)
```

Comment:
```go
fmt.Errorf("weaviate did not start up in %d seconds. Either the Weaviate URL %q is wrong or Weaviate did not start up in the interval given in 'startupTimeout'", startupTimeout, con.basePath)
```

Now that we are using `time.Duration` explicitly, we don't know for sure if the unit of time will necessarily be seconds. Luckily, `time.Duration` has a `String()` method, so this error message can now be:

```go
return fmt.Errorf("weaviate did not start up in %s. Either the Weaviate URL %q is wrong or Weaviate did not start up in the interval given in 'startupTimeout'", startupTimeout.String(), con.basePath)
```